framework/BaseClasses/BaseType.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Implements the common base class for all RAVEN objects (Entities and Interfaces)
+Assures access to:
+- InputData initialization
+- MessageHandler
+and similar utilities
+
+Created on Mar 16, 2013
+@author: crisr
+"""
+
+import sys
+
+#Internal Modules------------------------------------------------------------------------------------
+from utils import InputData, InputTypes, mathUtils
+from .MessageUser import MessageUser
+from .InputDataUser import InputDataUser
+import MessageHandler
+
+class BaseType(MessageUser, InputDataUser): # TODO add InputDataUser when all entities are converted
```

Comment:
I don't know that they both should in the future; I honestly think that those should be members of the `InputDataUser` class instead of `BaseType`, since they are strictly involved with reading the input params; however, the widespread custom modification of these methods in parts of the code make it difficult to narrow down.

I would much rather see these archetypes in place first, then see how we want to divide the activities between the two. I'm of the opinion that perhaps BaseEntity doesn't even have input specs; rather those specs all belong to the BaseInterface; however, with objects like Samplers and Distributions that appear to be both Entities and Interfaces, it's not clear yet who should control input reading.